Source/Android/jni/ButtonManager.cpp

```diff
@@ -333,7 +334,7 @@ void Init()
   }
   // Init our controller bindings
   IniFile ini;
-  ini.Load(File::GetUserPath(D_CONFIG_IDX) + std::string("Dolphin.ini"));
+  ini.Load(Paths::GetConfigDir() + std::string("Dolphin.ini"));
```

Comment:
Not only should that `std::string` cast be removed, it should be using `Paths::GetDolphinConfigFile()`!